### PR TITLE
Verify rosa is logged in to the correct OCM env, add billing-account to hashed log

### PR DESCRIPTION
**Pull request by rnetser**

### CODE REVIEW

**Comment by redhat-qe-bot2** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: rosa/cli.py [hunk 1]
```diff
@@ -72,11 +73,11 @@ def change_home_environment():
     os.environ["HOME"] = current_home
 
 
-def is_logged_in(aws_region=None, allowed_commands=None):
+def is_logged_in(aws_region=None, allowed_commands=None, env=None):
     _allowed_commands = allowed_commands or parse_help()
     try:
         res = build_execute_command(command="whoami", aws_region=aws_region, allowed_commands=_allowed_commands)
-        return "User is not logged in to OCM" not in res["err"]
+        return ("User is not logged in to OCM" not in res["err"]) and (res["out"]["OCM API"] == env)
```
Comment (myakove) [inline rosa/cli.py:80]: we need proper error for each, not logged in and logged in but to the wrong env.

**Review by myakove — CHANGES_REQUESTED** [review]

File: rosa/cli.py (hunk 1 above)
Comment (rnetser) [reply rosa/cli.py]: client behavior has changed; now it raises an exception.
updated the code

File: rosa/cli.py [hunk 2]
```diff
@@ -72,13 +76,20 @@ def change_home_environment():
...
+def is_logged_in(env, aws_region=None, allowed_commands=None):
     _allowed_commands = allowed_commands or parse_help()
+
     try:
         res = build_execute_command(command="whoami", aws_region=aws_region, allowed_commands=_allowed_commands)
-        return "User is not logged in to OCM" not in res["err"]
-    except CommandExecuteError:
-        return False
+
+        logged_in_env = res["out"].get("OCM API")
```
Comment (omrirh) [inline rosa/cli.py:85]:
```suggestion
        ocm_env = res["out"].get("OCM API")
```

File: rosa/cli.py [hunk 3]
```diff
@@ -72,13 +76,20 @@ def change_home_environment():
...
+        if logged_in_env != env:
```
Comment (omrirh) [inline rosa/cli.py:86]:
```suggestion
        if ocm_env != env:
```

File: rosa/cli.py [hunk 4]
```diff
@@ -72,13 +76,20 @@ def change_home_environment():
...
+            raise NotLoggedInOrWrongEnvError(
+                "User is logged in to OCM in {logged_in_env} environment and not {env} environment."
```
Comment (omrirh) [inline rosa/cli.py:88]:
```suggestion
                "User is logged in to OCM in {ocm_env} environment and not {env} environment."
```

File: rosa/cli.py (hunk 2 above)
Comment (rnetser) [reply rosa/cli.py]: changed to `logged_in_ocm_env` so it is clear that this is the logged in env

File: rosa/cli.py [hunk 5]
```diff
@@ -47,16 +49,19 @@ def hash_log_keys(log):
...
         return
 
-    build_execute_command(
-        command=f"login --env={env} --token={token}",
-        allowed_commands=_allowed_commands,
-    )
-    if not is_logged_in(allowed_commands=_allowed_commands, aws_region=aws_region):
-        raise NotLoggedInError("Failed to login to AWS.")
+    except NotLoggedInOrWrongEnvError as ex:
+        LOGGER.error(ex)
```
Comment (myakove) [inline rosa/cli.py:59]: No need for logger here since we expect to fail if not logged in, we will have error if line 64 fail

**Review by myakove — CHANGES_REQUESTED** [review]

**Review by myakove — APPROVED** [review]

**Pull request merged**